frontend/lib/src/components/widgets/NumberInput/NumberInput.tsx

```diff
@@ -380,7 +380,7 @@ export const NumberInput: React.FC<Props> = ({
         if (dirty) {
           commitValue({ value, source: { fromUi: true } })
         }
-        if (allowFormSubmitOnEnter) {
+        if (allowFormEnterToSubmit) {
           widgetMgr.submitForm(elementFormId, fragmentId)
         }
```

Comment:
I am wondering whether we should rather have a new parameter for `widgetMgr.submitForm` that takes a parameter `submitWay: "enter" | "submitButton"` or so which is passed to `submitForm`.  Then also the function 
`allowFormEnterToSubmit` could be removed and this part removed too:

```typescript
const allowFormEnterToSubmit = widgetMgr.allowFormEnterToSubmit(elementFormId)
```

because the `widgetMgr` can make the check internally. The advantage I see would be that the API is more obvious and probably less likely to forget this check when we develop new widgets in the future.
But not a super strong opinion, just wanted to float the idea 🙂

Reply:
That would work for the form submission check, but we also use the `allowFormEnterToSubmit` boolean to control `InputInstruction` display in the component